pkg/ruler/compat.go

```diff
@@ -102,10 +104,12 @@ func DefaultTenantManagerFactory(
 		logger log.Logger,
 		reg prometheus.Registerer,
 	) *rules.Manager {
+		evaluationDelay := overrides.EvaluationDelay(userID)
+
 		return rules.NewManager(&rules.ManagerOptions{
 			Appendable:      &PusherAppendable{pusher: p, userID: userID},
 			Queryable:       q,
-			QueryFunc:       engineQueryFunc(engine, q, cfg.EvaluationDelay),
+			QueryFunc:       engineQueryFunc(engine, q, evaluationDelay),
```

Comment:
I think by passing EvaluationDelay function (or overrides) to `engineQueryFunc`, we can make runtime updates to user's delay work. WDYT?

Reply:
Nice catch.